### PR TITLE
fix: translation of 1-js/06-advanced-functions/03-closure/

### PR DESCRIPTION
**Pull request by printfinn**
**目标章节**：1-js/06-advanced-functions/03-closure/

**当前上游最新 commit**：无
**本 PR 所做更改如下：**

文件名 | 参考上游 commit | 更改（理由）
-|-|-
article.md | 无 | 修改部分错误

### CODE REVIEW

File: 1-js/06-advanced-functions/03-closure/article.md [hunk 1]
```diff
@@ -303,7 +303,7 @@ let counter = makeCounter();
 ```smart header="闭包"
 开发者通常应该都知道“闭包”这个通用的编程术语。
 
-[闭包](https://en.wikipedia.org/wiki/Closure_(computer_programming)) 一个记住其外部变量并可以访问这些变量的函数。在某些编程语言中，这是不可能的，或者应该以一种特殊的方式编写函数来实现。但是如上所述，在 JavaScript 中，所有函数都是天生闭包的（只有一个例外，将在 <info:new-function> 中讲到）。
+[闭包](https://en.wikipedia.org/wiki/Closure_(computer_programming)) 是一个记住其外部变量并可以访问这些变量的函数。在某些编程语言中，这是不可能的，或者应该以一种特殊的方式编写函数来实现。但是如上所述，在 JavaScript 中，所有函数都是天生闭包的（只有一个例外，将在 <info:new-function> 中讲到）。
```
Comment (leviding) [inline 1-js/06-advanced-functions/03-closure/article.md:306]:
```suggestion
[闭包](https://en.wikipedia.org/wiki/Closure_(computer_programming)) 是指函数所拥有的可以记住其外部变量并可以访问这些变量特性。在某些编程语言中，这是不可能的，或者应该以一种特殊的方式编写函数来实现。但如上所述，在 JavaScript 中，所有函数都是天生闭包的（只有一个例外，将在 <info:new-function> 中讲到）。
```

Comment (leviding) [reply 1-js/06-advanced-functions/03-closure/article.md:306]: Closure is a technique, is a feature, not a function.

In wikipedia: In [programming languages](https://en.wikipedia.org/wiki/Programming_language), a closure, also lexical closure or function closure, is a technique for implementing [lexically scoped](https://en.wikipedia.org/wiki/Lexically_scoped) [name binding](https://en.wikipedia.org/wiki/Name_binding) in a language with [first-class functions](https://en.wikipedia.org/wiki/First-class_function). [Operationally](https://en.wikipedia.org/wiki/Operational_semantics), a closure is a [record](https://en.wikipedia.org/wiki/Record_(computer_science)) storing a [function](https://en.wikipedia.org/wiki/Function_(computer_science))[[a]](https://en.wikipedia.org/wiki/Closure_(computer_programming)#cite_note-1) together with an environment.[[1]](https://en.wikipedia.org/wiki/Closure_(computer_programming)#cite_note-2) The environment is a mapping associating each [free variable](https://en.wikipedia.org/wiki/Free_variable) of the function (variables that are used locally, but defined in an enclosing scope) with the [value](https://en.wikipedia.org/wiki/Value_(computer_science)) or [reference](https://en.wikipedia.org/wiki/Reference_(computer_science)) to which the name was bound when the closure was created.[[b]](https://en.wikipedia.org/wiki/Closure_(computer_programming)#cite_note-3) Unlike a plain function, a closure allows the function to access those captured variables through the closure's copies of their values or references, even when the function is invoked outside their scope.

Link: https://en.wikipedia.org/wiki/Closure_(computer_programming)

---

In MDN: Closure is the binding which defines the [scope](https://developer.mozilla.org/en-US/docs/Glossary/Scope) of execution. In [JavaScript](https://developer.mozilla.org/en-US/docs/Glossary/JavaScript), [functions](https://developer.mozilla.org/en-US/docs/Glossary/Function) create a closure context.

Link: https://developer.mozilla.org/en-US/docs/Glossary/Closure

---

Stackoverflow: https://stackoverflow.com/questions/50529400/is-the-definition-of-javascript-closures-on-mdn-wrong

Comment (leviding) [reply 1-js/06-advanced-functions/03-closure/article.md:306]: > 我不确定这是不是闭包的定义, 我觉得闭包指的是函数和函数的binding的一个组合, 不仅仅是一个特性?

说是组合没毛病，但上面你那个翻译说的是 闭包是...函数，这个不对。

是特性，我觉得也是一个说法，当然以 MDN 为准更权威。

Comment (printfinn) [reply 1-js/06-advanced-functions/03-closure/article.md:306]: 您发的链接我都仔细看了, 我也觉得直接把closure定义为函数欠妥, 从英文原文看, 作者说的“A closure is a function that remembers its outer variables and can access them.”确实是定义为了一个“可以...的函数”. 我们不妨直译原文, 而后加个译注? 因为我对JS的闭包其实不太了解, 没有资格来决定到底应该翻译成什么, 您可以按照您觉得比较合适的方式帮我修改一下, 或者直接commit您之前修改的翻译也行.

Comment (leviding) [reply 1-js/06-advanced-functions/03-closure/article.md:306]: @printfinn @songhn233 我优化了一下闭包的说法，你们看看怎样？是否有啥修改意见和建议

Comment (printfinn) [reply 1-js/06-advanced-functions/03-closure/article.md:306]: 我觉得这个很行

**Pull request merged**